### PR TITLE
Use DetailsPlusPayments for wallet refresh

### PR DESCRIPTION
**Pull request by patrickxb**
Uses the new endpoint for refreshing wallet state data.

Some more minor tweaks from linters...

### CODE REVIEW

File: go/stellar/relays/relays.go [hunk 1]
```diff
@@ -141,7 +141,7 @@ func encrypt(relay stellar1.RelayContents, encryptFor keybase1.TeamApplicationKe
 		return res, err
 	}
 	secbox := secretbox.Seal(
-		nil, clearpack[:], &nonce, (*[32]byte)(&encryptFor.Key))
+		nil, clearpack, &nonce, (*[32]byte)(&encryptFor.Key))
```
Comment (mlsteele) [inline go/stellar/relays/relays.go]: did you use some tool or just an eagle eye for these?

File: go/stellar/wallet_state.go [hunk 2]
```diff
@@ -407,8 +410,12 @@ func (w *WalletState) SubmitRelayClaim(ctx context.Context, post stellar1.RelayC
 // MarkAsRead is an override of remoter's MarkAsRead.
 func (w *WalletState) MarkAsRead(ctx context.Context, accountID stellar1.AccountID, mostRecentID stellar1.TransactionID) error {
 	err := w.Remoter.MarkAsRead(ctx, accountID, mostRecentID)
-	mctx := libkb.NewMetaContext(ctx, w.G())
-	w.Refresh(mctx, accountID, "MarkAsRead")
+	if err == nil {
+		mctx := libkb.NewMetaContext(ctx, w.G())
+		if rerr := w.Refresh(mctx, accountID, "MarkAsRead"); rerr != nil {
```
Comment (mlsteele) [inline go/stellar/wallet_state.go]: should this kind of thing by background/async? you'd know better whether this function returning is on any latency critical path

File: go/stellar/relays/relays.go (hunk 1 above)
Comment (patrickxb) [reply go/stellar/relays/relays.go]: some linter found it...i can't remember which one, i was trying out a bunch.

File: go/stellar/wallet_state.go (hunk 2 above)
Comment (patrickxb) [reply go/stellar/wallet_state.go]: good idea...i'll change it.

**Pull request merged**